File: 217-ContainsDuplicate/217-ContainsDuplicate.py
# Last updated: 8/23/2025, 3:51:50 PM
class Solution(object):
    def containsDuplicate(self, nums):
        """
        :type nums: List[int]
        :rtype: bool
        """
        # Brute force
        # for i in range(len(nums)):
        #     for j in range(i + 1, len(nums)):
        #         if nums[i] == nums[j]:
        #             return True
        #     return False        
        # Time: O(n^2) and Space O(1)
        # Optimal 
        return True if len(set(nums)) < len(nums) else False

File: 217-ContainsDuplicate/217-ContainsDuplicate.cpp
// Last updated: 8/23/2025, 3:51:51 PM
class Solution {
public:
    bool containsDuplicate(vector<int>& nums) {
        
        int check=5;
        sort(nums.begin(),nums.end());
        for(int i=0;i<nums.size();i++)
        {
            int find=nums[i];
             nums.erase(nums.begin()+i);
                 if(binary_search(nums.begin(), nums.end(), find))
                     return true;
        }
        return false;
    }
};